torch/csrc/CudaIPCTypes.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+#include <c10/core/Allocator.h>
+
+#define CUDA_IPC_REF_COUNTER_FILE_SIZE 10000
+#define CUDA_IPC_WARN_AFTER_X_BLOCKS_IN_LIMBO 1000
+
+namespace torch {
+
+
+struct CudaIPCReceivedData {
+  CudaIPCReceivedData(std::shared_ptr<void> shared_ptr);
+  std::shared_ptr<void> shared_ptr_;
+};
+
+struct CudaIPCSentData {
+  CudaIPCSentData(std::string handle, int64_t offset, int64_t* counter_ptr)
+      : handle(handle), offset(offset), counter_ptr(counter_ptr){};
+  ~CudaIPCSentData();
+  int64_t get();
+  std::string handle;
+  int64_t offset;
+  int64_t* counter_ptr; // Reference counter shared memory block
```

Comment:
nit: Members in C++ should have a trailing underscore. (I understand we have some legacy code that doesn't abide by this; let's keep this up for new code.)